autometa/binning/summary.py

```diff
@@ -0,0 +1,364 @@
+    for length in df.length.sort_values(ascending=False):
+        lengths += length
+        if lengths > target_size:
+            return length
+
+
+def get_metabin_stats(bin_df, markers_fpath, assembly):
+    """Retrieve statistics for all clusters recovered from Autometa binning.
+
+    Parameters
+    ----------
+    bin_df : pd.DataFrame
+        Autometa binning table. index=contig, cols=['cluster','length', 'GC', 'coverage', ...]
+    markers_fpath : str
+        </path/to/{domain}.markers.tsv
+    assembly : str
+        </path/to/corresponding/metagenome/assembly.fna
+
+    Returns
+    -------
+    pd.DataFrame
+        dataframe consisting of various metabin statistics indexed by column.
+    """
+    stats = []
+    markers_df = Markers.load(markers_fpath)
+    for cluster, dff in bin_df.fillna(value={"cluster": "unclustered"}).groupby(
+        "cluster"
+    ):
+        contigs = set(dff.index)
+        length_weighted_coverage = np.average(
+            a=dff.coverage, weights=dff.length / dff.length.sum()
+        )
+        length_weighted_gc = np.average(a=dff.GC, weights=dff.length / dff.length.sum())
+        num_expected_markers = markers_df.shape[1]
+        pfam_counts = markers_df.loc[markers_df.index.isin(dff.index)].sum()
+        if pfam_counts.empty:
+            total_markers = 0
+            num_single_copy_markers = 0
+            num_markers_present = 0
+            completeness = pd.NA
+            purity = pd.NA
+        else:
+            total_markers = pfam_counts.sum()
+            num_single_copy_markers = pfam_counts[pfam_counts == 1].count()
+            num_markers_present = pfam_counts[pfam_counts >= 1].count()
+            completeness = num_markers_present / num_expected_markers * 100
+            purity = num_single_copy_markers / num_markers_present * 100
+
+        stats.append(
+            {
+                "cluster": cluster,
+                "nseqs": dff.shape[0],
+                "seqs_pct": dff.shape[0] / bin_df.shape[0] * 100,
+                "size (bp)": dff.length.sum(),
+                "size_pct": dff.length.sum() / bin_df.length.sum() * 100,
+                "N90": fragmentation_metric(dff, quality_measure=0.9),
+                "N50": fragmentation_metric(dff, quality_measure=0.5),
+                "N10": fragmentation_metric(dff, quality_measure=0.1),
+                "length_weighted_gc": length_weighted_gc,
+                "min_GC": dff.GC.min(),
+                "max_GC": dff.GC.max(),
+                "length_weighted_coverage": length_weighted_coverage,
+                "min_coverage": dff.coverage.min(),
+                "max_coverage": dff.coverage.max(),
+                "completeness": completeness,
+                "purity": purity,
+                "num_total_markers": total_markers,
+                f"num_unique_markers (expected {num_expected_markers})": num_markers_present,
+                "num_single_copy_markers": num_single_copy_markers,
+            }
```

Comment:
I think the current version also reports standard deviation of coverage and GC? I'm not sure how you do that in a weighted fashion, however. This also relates to issue #46

Reply:
Should be easy enough to add if you will accept `dff.GC.std()` and `dff.coverage.std()`